service/history/workflow/update/update.go

```diff
@@ -87,9 +87,14 @@ type (
 	// (OnMessage calls) must be done while holding the workflow lock.
 	Update struct {
 		// accessed only while holding workflow lock
-		id              string
-		state           state
-		request         *anypb.Any // of type *updatepb.Request, nil when not in stateAdmitted or stateSent.
+		id    string
+		state state
+		// request is nil when:
+		//   - update is in stateAccepted,
+		//   - update was restored in stateAdmitted with newAdmitted from UpdateInfo.AdmissionInfo.
+		//       We don't load request form event because if there is durable UpdateAdmitted event with request,
+		//       then we don't write this request 2nd time to UpdateAccepted event.
+		request         *anypb.Any // of type *updatepb.Request
```

Comment:
```suggestion
		// The `request` field holds the update payload submitted with the original request. It is stored in the
		// registry in order to be written to history in an UpdateAccepted event.
		// Therefore, it is nil when the update in the registry is in stateAccepted, stateCompleted etc, since then the
		// request has already been written to an UpdateAccepted event.
		// In addition, it is nil when the update in the registry is in stateAdmitted AND it was created from an
		// UpdateInfo.AdmissionInfo entry in MutableState. In this case, the reason it is nil is the following:
		// 1. The presence of the AdmissionInfo entry in MutableState implies that there is an UpdateAdmitted event in
		//    history. This event always contains the original request payload.
		// 2. Therefore, it is not necessary to write a second copy of the request payload to an UpdateAccepted event,
		//    so we don't *need* to load the request into the registry.
		// 3. Furthermore, it is possible that many UpdateAdmitted events were created after a Reset or during conflict
		//    resolution. In that situation we *must not* attempt to load all the payloads into the registry.
```

Reply:
I did one little correction. It is is also used to be actually sent to the worker. Server put in the WFT when WFT is started, which is different place from where WFT is scheduled (even if it is speculative).